src/v/kafka/server/quota_manager.cc

```diff
@@ -79,13 +83,7 @@ ss::future<> quota_manager::start() {
 }
 
 ss::future<clock::duration> quota_manager::maybe_add_and_retrieve_quota(
-  std::optional<std::string_view> quota_id,
-  clock::time_point now,
-  quota_mutation_callback_t cb) {
-    // requests without a client id are grouped into an anonymous group that
-    // shares a default quota. the anonymous group is keyed on empty string.
-    auto qid = quota_id.value_or("");
-
+  tracker_key qid, clock::time_point now, quota_mutation_callback_t cb) {
```

Comment:
observation: I don't see any corresponding usage changes in this commit, which is fine in terms of variant usage, but I note that we're moving from a reference type to a value type. 

question: Have we introduced a string copy at the call site, and is that of any concern?

Reply:
Yeah, this was a coroutine param taking a reference so I've opted for a copy here instead of passing by reference.

I was wondering if it would be worth allowing coroutine pass-by-reference inside `quota_manager` more widely to prevent copies but so far I went with the safer and simpler option of just copying. @BenPope wdyt?

Reply:
Yeah, if the original ref-based formulation was unsafe, I'd definitely stick with the value param. Looking down into this function, it looks like we take another copy into the lambda capture, then another into `add_quota_id`. Presumably these could be moves?

Reply:
We need to copy into the lambda capture here, because `qid` is used further down: https://github.com/redpanda-data/redpanda/blob/25b53d7b8f753065c414cdef520d55be0271fe78/src/v/kafka/server/quota_manager.cc#L78-L80

Inside the lambda, we could move the argument passed to `add_quota_id`. Let me fix that in a follow-up PR.